apps/payments/next/app/[locale]/[offeringId]/[interval]/upgrade/[cartId]/layout.tsx

```diff
@@ -24,30 +24,67 @@ export default async function UpgradeLayout({
   children: React.ReactNode;
   params: CheckoutParams;
 }) {
-  // Temporarily defaulting to `accept-language`
-  // This to be updated in FXA-9404
-  //const locale = getLocaleFromRequest(
-  //  params,
-  //  headers().get('accept-language')
-  //);
-  const locale = headers().get('accept-language') || DEFAULT_LOCALE;
+  const { locale } = params;
+  const acceptLanguage = headers().get('accept-language');
+  const l10n = getApp().getL10n(acceptLanguage, locale);
 
   const cartDataPromise = getCartAction(params.cartId);
   const cmsDataPromise = fetchCMSData(params.offeringId, locale);
-  const l10n = getApp().getL10n(locale);
   const [cms, cart] = await Promise.all([cmsDataPromise, cartDataPromise]);
   const purchaseDetails =
     cms.defaultPurchase.purchaseDetails.localizations.at(0) ||
     cms.defaultPurchase.purchaseDetails;
 
-  assert(cart.fromOfferingConfigId, 'fromOfferingConfigId is missing in cart');
-  assert(cart.fromPrice, 'fromPrice is missing in cart');
-
-  const currentCmsDataPromise = fetchCMSData(cart.fromOfferingConfigId, locale);
-  const currentCms = await currentCmsDataPromise;
-  const currentPurchaseDetails =
-    currentCms.defaultPurchase.purchaseDetails.localizations.at(0) ||
-    currentCms.defaultPurchase.purchaseDetails;
+  const displayPurchaseDetails = async () => {
```

Comment:
This seems to be a new component and should be defined as one. It can remain in this layout.tsx file, but it should be it's own component.

Alternatively, instead of creating a new component, since it's only the success page that needs a different layout, a separate `layout.tsx` could be defined for the success page.

Either option works.